packages/convertim/src/Model/Order/ConvertimOrderDataToOrderDataMapper.php

```diff
@@ -0,0 +1,190 @@
+            $orderData->city = $convertimBillingAddressData->getCity();
+            $orderData->postcode = $convertimBillingAddressData->getPostcode();
+            $orderData->country = $this->getCountryByName($convertimBillingAddressData->getCountry());
+        } else {
+            $orderData->companyName = $convertimDeliveryAddressData->getCompanyName();
+            $orderData->street = $convertimDeliveryAddressData->getStreet();
+            $orderData->city = $convertimDeliveryAddressData->getCity();
+            $orderData->postcode = $convertimDeliveryAddressData->getPostcode();
+            $orderData->country = $this->getCountryByName($convertimDeliveryAddressData->getCountry());
+        }
+
+        $orderData->deliveryAddressSameAsBillingAddress = $convertimOrderData->isBillingAddressSameAsDeliveryAddress();
+
+        if (!$convertimOrderData->isBillingAddressSameAsDeliveryAddress()) {
+            $orderData->deliveryFirstName = $convertimDeliveryAddressData->getName();
+            $orderData->deliveryLastName = $convertimDeliveryAddressData->getLastName();
+            $orderData->deliveryCompanyName = $convertimDeliveryAddressData->getCompanyName();
+            $orderData->deliveryTelephone = $convertimDeliveryAddressData->getCurrierTelephoneNumberWithPrefix();
+            $orderData->deliveryStreet = $convertimDeliveryAddressData->getStreet();
+            $orderData->deliveryCity = $convertimDeliveryAddressData->getCity();
+            $orderData->deliveryPostcode = $convertimDeliveryAddressData->getPostCode();
+            $orderData->deliveryCountry = $this->getCountryByName($convertimDeliveryAddressData->getCountry());
+        }
+        $orderData->note = $convertimOrderData->getNote();
+
+        $this->mapProducts($convertimOrderData, $orderData);
+        $this->mapConvertimTransportOrPaymentDataToOrderItem($convertimOrderData->getPaymentData(), $orderData);
+        $this->mapConvertimTransportOrPaymentDataToOrderItem($convertimOrderData->getTransportData(), $orderData);
+
+        $this->cartFacade->deleteCart($cart);
+
+        return $orderData;
+    }
+
+    /**
+     * @param string $uuid
+     * @param \Shopsys\FrameworkBundle\Model\Order\Item\OrderItemData[] $orderItemDatas
+     * @return \Shopsys\FrameworkBundle\Model\Order\Item\OrderItemData
+     */
+    protected function getOrderItemByUuid(string $uuid, array $orderItemDatas): OrderItemData
+    {
+        foreach ($orderItemDatas as $orderItemData) {
+            if ($orderItemData->getItemUuid() === $uuid) {
+                return $orderItemData;
+            }
+        }
+
+        throw new OrderItemNotFoundException(sprintf('Order item with uuid "%s" not found.', $uuid));
+    }
+
+    /**
+     * @param string|null $countryName
+     * @return \Shopsys\FrameworkBundle\Model\Country\Country
+     */
+    protected function getCountryByName(?string $countryName): Country
```

Comment:
this is quite complicated and still can end up with an error (when there is no country enabled on the domain) :thinking: Moreover, does Convertim really return the country name? Based on the examples in the docs, it seems like country ISO code, or?

I would rather add a new method to the country repository - `getByCode(string $countryCode)` and throw an exception when the country is not found